File: week1/p4.py
arr = [1, 2, 0, 4, 3, 0, 5, 0]

count = 0  # position for next non-zero element
for i in range(len(arr)):
    if arr[i] != 0:
        arr[count] = arr[i]
        count += 1

while count < len(arr):
    arr[count] = 0
    count += 1

print(arr)

File: week1/p4.c
#include <stdio.h>

int main() {
    int arr[] = {1, 2, 0, 4, 3, 0, 5, 0};
    int n = sizeof(arr) / sizeof(arr[0]);

    int count = 0; // position for next non-zero element

    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) {
            arr[count++] = arr[i];
        }
    }
    while (count < n) {
        arr[count++] = 0;
    }

    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    return 0;
}
